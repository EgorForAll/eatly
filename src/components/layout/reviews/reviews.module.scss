.reviews {
  width: 100%;
}

.container {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto;
  align-items: center;
  padding: 77px 0 70px 115px;

  @media (max-width: 992px) {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: repeat(3, auto);
    padding: 20px;
    overflow: hidden;
  }
}

.title {
  font-family: "Poppins", sans-serif;
  font-weight: 600;
  font-size: 52px;
  line-height: 57%;
  color: #000;
  text-align: center;
  margin-top: 119px;
  letter-spacing: -3px;

  span {
    color: #6C5FBC;
  }

  @media (max-width: 992px) {
    font-size: 30px;
    margin-top: 0;
    padding-top: 20px;
    margin-bottom: 50px;
  }
}

.fullReview {
  grid-column: 1/2;
  grid-row: 1/4;
  margin-right: 44px;

  @media (max-width: 992px) {
    width: 90%;
    grid-row: 1/2;
    grid-column: 1/2;
    margin-right: 0;
    margin-bottom: 20px;
  }
}


.shortReviewsList {
  grid-column: 2/3;
  grid-row: 1/2;
  overflow-x: scroll;
  display: grid;
  grid-template-columns: repeat(5, 500px);
  grid-template-rows: 1fr;
  column-gap: 20px;
  row-gap: 30px;
  height: 160px;

  //Scrollbar
  &::-webkit-scrollbar {
    height: 12px;
  }

  &::-webkit-scrollbar-track {
    border-radius: 5px;
    background-color: #CECCDE;
    margin-right: 100px;

    @media (max-width: 992px) {
      margin-right: 0;
    }
  }

  &::-webkit-scrollbar-track:hover {
    background-color: #cecbde;
  }

  &::-webkit-scrollbar-track:active {
    background-color: #cecbde;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 5px;
    background-color: #675BB1;
  }

  &::-webkit-scrollbar-thumb:hover {
    background-color: #675BB1;
  }

  &::-webkit-scrollbar-thumb:active {
    background-color: #675BB1;
  }



  @media (max-width: 992px) {
    grid-column: 1/2;
    grid-row: 2/3;
    height: auto;
    column-gap: 0;
    row-gap: 10px;
    grid-template-columns: repeat(5, 300px);
    margin-bottom: 30px;
  }
}

.notFound {
  font-family: "Poppins", sans-serif;
  font-weight: 600;
  font-size: 35px;
  color: #6C5FBC;
  text-align: center;
}



