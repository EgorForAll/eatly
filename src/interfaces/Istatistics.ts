export interface IStats {
    title: string;
    description: string;
}
