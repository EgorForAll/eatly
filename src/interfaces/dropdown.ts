export interface IDropdown {
    title: string;
    description: string;
}