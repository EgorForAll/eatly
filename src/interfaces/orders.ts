export interface IOrders {
    name: string;
    time: string;
    status: string;
    image: string;
    imageWebp: string;
    is_fail?: boolean;
}