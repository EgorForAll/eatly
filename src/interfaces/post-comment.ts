export interface IPostComment {
    body: string;
    postId: string | undefined;
    userId: number;
}