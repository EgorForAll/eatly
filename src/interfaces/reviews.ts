export interface IReview {
    name?: string;
    image?: string;
    image_webp?: string;
    text: string;
    time?: string;
    rate: number;
}