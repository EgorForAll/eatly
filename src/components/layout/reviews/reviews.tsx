// @flow
import * as React from "react";
import styles from "./reviews.module.scss"
import FullReview from "@/components/blocks/full-review/full-review";
import { IReview } from "@/interfaces/reviews";
import ShortReview from "@/components/blocks/short-review/short-review";
import { CSSProperties, useEffect, useState } from "react";
import axios from "axios";
import { detectFirefox } from "@/utils/utils";

const Reviews: React.FC = () => {
  const [reviews, setReviews] = useState<IReview[] | []>([]);

  useEffect(() => {
    axios
      .get("https://dummyjson.com/comments")
      .then(({ data }) => setReviews(data.comments))
      .catch((e) => console.error(e));
  }, []);

  const isFirefox = detectFirefox();
  const scrollStyle: CSSProperties = {
    scrollbarWidth: "thin",
    scrollMargin: "0 100px 0 0",
    scrollbarColor: "#675BB1 #cecbde",
  };

  const lastReviews = reviews.slice(1, 6);
  const firstReview = reviews[0];

  return (
    <section className={styles.reviews}>
      <h2 className={styles.title}>
        <span>Customer</span> Say
      </h2>
      <div className={styles.container}>
        {reviews.length > 0 ? (
          <>
            <div className={styles.fullReview}>
              {firstReview && <FullReview review={firstReview} />}
            </div>
            <ul
              className={styles.shortReviewsList}
              style={isFirefox ? scrollStyle : undefined}
            >
              {lastReviews &&
                lastReviews.map((item, index) => (
                  <ShortReview text={item.body} key={index} />
                ))}
            </ul>
          </>
        ) : <h3 className={styles.notFound}>Отзывы не найдены</h3>}
      </div>
    </section>
  );
};

export default Reviews;
