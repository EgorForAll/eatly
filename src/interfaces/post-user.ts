import {IUser} from "@/interfaces/user";

export interface IPostUser {
    postId: number;
    user: IUser;
}