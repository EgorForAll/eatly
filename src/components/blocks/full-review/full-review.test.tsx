import {screen, render} from "@testing-library/react";
import FullReview from "./full-review";
import '@testing-library/jest-dom'
import {IReview} from "@/interfaces/reviews";

const mockReview: IReview = {
    id: 1,
    body: 'Some long test from some strange person',
    postId: 2,
    user: {
        id: 2,
        username: 'Marko'
    }
}

test('Full review renders correctly', () => {
    render(<FullReview review={mockReview}/>)
    const review = screen.getByTestId('review')
    expect(review).toMatchSnapshot()
})