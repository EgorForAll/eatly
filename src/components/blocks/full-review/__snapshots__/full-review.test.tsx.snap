// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full review renders correctly 1`] = `
<div
  class="review"
  data-testid="review"
>
  <header
    class="header"
  >
    <div
      class="clientInfo"
    >
      <h2
        class="name"
      >
        @
        Marko
      </h2>
    </div>
    <div
      class="quotes"
    >
      <svg>
        quotes.svg
      </svg>
    </div>
  </header>
  <p
    class="text"
  >
    “
    Some long test from some strange person
    ”
  </p>
</div>
`;
